force-app/main/default/classes/PS_CommitmentRequest.cls

```diff
@@ -267,8 +267,11 @@ public inherited sharing class PS_CommitmentRequest {
             : UserInfo.getDefaultCurrency();
 
         if (isElevateScheduleFieldsChanged(rd, oldRd)) {
+            // If the schedule has a Next Payment Amount, we need to send it
+            List<RecurringDonationSchedule__c> rdSchedules = scheduleService.getExistingSchedules(rd);
+            Decimal nextPaymentAmount = rdSchedules.size() > 0 ? rdSchedules[0].NextPaymentAmount__c : null;
```

Comment:
I'd be happier if we didn't have any implementation details in this class.  May getNextPaymentAmount in RD2_ScheduleService??

Also, is it safe to assume the amount is on the first active schedule? Will it be on both schedules for a 1st and 15th cadence?

Reply:
I didn't realize 1st and 15th used 2 Active schedules, I'll double check that they're in the correct sort order. I can create that method, good suggestion.